src/VisualStudio/Core/SolutionExplorerShim/AnalyzersCommandHandler.cs

```diff
@@ -96,81 +113,199 @@ public void Initialize(IMenuCommandService menuCommandService)
             }
         }
 
-        private MenuCommand AddCommandHandler(IMenuCommandService menuCommandService, int roslynCommand, EventHandler handler)
+        public IContextMenuController AnalyzerFolderContextMenuController
         {
-            var commandID = new CommandID(Guids.RoslynGroupId, roslynCommand);
-            var menuCommand = new MenuCommand(handler, commandID);
-            menuCommandService.AddCommand(menuCommand);
+            get
+            {
+                if (_analyzerFolderContextMenuController == null)
+                {
+                    _analyzerFolderContextMenuController = new ContextMenuController(
+                        ID.RoslynCommands.AnalyzerFolderContextMenu,
+                        ShouldShowAnalyzerFolderContextMenu,
+                        UpdateAnalyzerFolderContextMenu);
+                }
 
-            return menuCommand;
+                return _analyzerFolderContextMenuController;
+            }
+        }
+
+        private bool ShouldShowAnalyzerFolderContextMenu(IEnumerable<object> items)
+        {
+            return items.Count() == 1;
```

Comment:
return !items.Skip(1).Any(); ?

doesn't matter if it is expected to have small number of items though..